### ISSUE TITLE
ESQL: date_extract using a constant/literal chrono_field fails to record the warning

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#100038]
### Description

For query `from test | eval x = date_extract(pattern, date.date) | keep x` where `pattern` is in the index set as `y123ear` (a non-existent ChronoField value) will return `x` as `null` in the results and have the response record two warnings in headers one of which is `"java.lang.IllegalArgumentException: No enum constant java.time.temporal.ChronoField.Y123EAR"`

But, if the same non-existent value `y123ear` is used as a constant ie `from test | eval x = date_extract(\"y123year\", date.date) | keep x` results in an error message and no warning recorded:

```
{
    "error": {
        "root_cause": [
            {
                "type": "esql_illegal_argument_exception",
                "reason": "invalid date field for [date_extract(\"y123year\", date.date)]: y123year",
                "suppressed": [
                    {
                        "type": "task_cancelled_exception",
                        "reason": "parent task was cancelled [cancelled]"
                    },
                    {
                        "type": "task_cancelled_exception",
                        "reason": "parent task was cancelled [cancelled]"
                    }
                ]
            }
        ],
        "type": "esql_illegal_argument_exception",
        "reason": "invalid date field for [date_extract(\"y123year\", date.date)]: y123year",
        "suppressed": [
            {
                "type": "task_cancelled_exception",
                "reason": "parent task was cancelled [cancelled]"
            },
            {
                "type": "task_cancelled_exception",
                "reason": "parent task was cancelled [cancelled]"
            }
        ]
    },
    "status": 500
}
```

**Comment by astefan** [comment]
We'll discuss this or a future improvement, if considered to be something we must do (still do a warning even if the chrono value comes from a constant/literal). There is a slight difference in usage here, as one can argue that a constant/literal is a value that an user is on purpose writing/using in the query and the user must be pro-actively notified he/she did something wrong when authored the query. BUT, at the same time that constant can come from another series of deeply nested functions that also act on constants themselves. Also, this behavior is greatly inconsistent.

**Comment by alex-spies** [comment]
A warning vs exception inconsistency also occurs when folding constants, where we normally return a warning if e.g. an arithmetic expression made from literals overflows, but for date math overflows we return a `VerificationException` (see https://github.com/elastic/elasticsearch/issues/99758).

IMO an exception is better if constants/literals cause the problem, as that means the whole query is faulty and we should not continue executing it. Agree that we should discuss and tackle in the future.

**Comment by wchaparro** [comment]
@fang-xing-esql should this still be assigned to you?

**Comment by fang-xing-esql** [comment]
> [@fang-xing-esql](https://github.com/fang-xing-esql) should this still be assigned to you?

@wchaparro Yep, I'll keep it in my queue unless someone else has the bandwidth to fix it before I get to it.
